### PR TITLE
removing template from parsed file model - step 1 - flow template

### PR DESCRIPTION
**Pull request by vinkobuble**
Each model will be refactored in the same manner until `template_model` is removed from the `ParsedFyPyFile` base class.

### CODE REVIEW

File: library/src/fy_library/domain/parsed_fy_py_file.py [hunk 1]
```diff
@@ -45,7 +47,15 @@ class ParsedFyPyFile(FyPyFileParts):
...
-    template_model: FlowTemplateModel
+    flow_name: PythonEntityName
+    return_type: str
+    properties: List[PropertyMixinModel]
+    methods: List[MethodMixinModel]
+
+    @computed_field
+    @property
+    def entity_key(self) -> str:
+        return self.flow_name.snake_case
```
Comment (vinkobuble) [inline library/src/fy_library/domain/parsed_fy_py_file.py:58]: copied from the `FlowTemplateModel` class.

File: library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/flow_template_model_fy.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+from fy_library.domain.fy_py_template_models import FlowTemplateModelWithPropertySetters
+
+
+flow FlowTemplateModelWithPropertySetters_UsingParsedFyPyFileAndPropertySettersTemplateModels -> FlowTemplateModelWithPropertySetters:
+    property parsed_fy_py_file using setter
+    property parsed_fy_py_files_map_by_key using setter
+    property property_mixins using template_model_properties
+    property property_setter_mixins using property_mixins
+    property parsed_fy_py_files using property_setter_mixins__mapped_to_abstract_property
+    property template_models using parsed_fy_py_files__with_property_setters
```
Comment (vinkobuble) [inline library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/flow_template_model_fy.py:14]: This still needs to be used until at least `AbstractProperty` is migrated.

File: library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/flow_template_model_fy.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    ):
+        self._parsed_fy_py_file = parsed_fy_py_file
+        self._parsed_fy_py_files_map_by_key = parsed_fy_py_files_map_by_key
+        super().__init__(*args, **kwargs)
+
+    def __call__(self) -> FlowTemplateModelWithPropertySetters:
+        # fy:end <<<===
+        parsed_flow_fy_py_file = self._parsed_fy_py_file
+        assert isinstance(parsed_flow_fy_py_file, ParsedFlowFyPyFile)
+        return FlowTemplateModelWithPropertySetters(
```
Comment (vinkobuble) [inline library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/flow_template_model_fy.py:71]: These flows will create a complete new template object. We'll rename this class to `FlowTemplateModel` once we can remove other usage of `FlowTemplateModel`.

File: library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/main_fy.py [hunk 4]
```diff
@@ -78,8 +76,14 @@ def __init__(
 
     def __call__(self) -> BaseTemplateModel:
         # fy:end <<<===
-        return (
-            self._optional_base_flow_template_model_with_property_setters
-            or self._optional_flow_template_model_with_property_setters
-            or self._parsed_fy_py_file.template_model
-        )
+        match self._parsed_fy_py_file.file_type:
```
Comment (vinkobuble) [inline library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/main_fy.py:79]: The main flow will only have a `match/case` branching to invoke particular flow.

File: library/src/fy_library/mixins/property/mixin_imports/using_parsed_fy_py_file_fy.py [hunk 5]
```diff
@@ -58,25 +59,25 @@ def _mixin_imports(self) -> List[str]:
...
                     method_mixins=cast(
                         ParsedFlowFyPyFile, self._parsed_fy_py_file
-                    ).template_model.methods,
+                    ).methods,
                 )()
             case ParsedFyPyFileKind.BASE_FLOW:
                 return BaseFlowImports_Flow(
                     property_mixins=cast(
-                        ParsedFlowFyPyFile, self._parsed_fy_py_file
+                        ParsedBaseFlowFyPyFile, self._parsed_fy_py_file
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/mixin_imports/using_parsed_fy_py_file_fy.py:72]: This was an omission in the implementation.

File: library/src/fy_library/mixins/property/parsed_flow_fy_py_file/using_parsed_fy_py_file_fy.py [hunk 6]
```diff
@@ -69,6 +69,11 @@ def _parsed_flow_fy_py_file(self) -> ParsedFlowFyPyFile:
             post_marker_file_content=self._post_marker_file_content,
             file_path=self._fy_py_file_to_parse,
             user_imports=self._flow_file_split.user_imports,
+            flow_name=flow_name,
+            return_type=self._flow_file_split.return_type,
+            properties=self._included_mixins.property_mixins,
+            methods=self._included_mixins.method_mixins,
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/parsed_flow_fy_py_file/using_parsed_fy_py_file_fy.py:75]: Copied over from the template instantiation code.

File: library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/flow_template_model_fy.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+"""fy
+from fy_library.domain.fy_py_template_models import FlowTemplateModelWithPropertySetters
+
+
+flow FlowTemplateModelWithPropertySetters_UsingParsedFyPyFileAndPropertySettersTemplateModels -> FlowTemplateModelWithPropertySetters:
```
Comment (PerisicJere) [inline library/src/fy_library/flows/create_template_model_using_parsed_fy_py_file/flow_template_model_fy.py:8]:
```suggestion
flow FlowTemplateModelWithPropertySetters_UsingParsedFyPyFileAndPropertySettersTemplateModels -> FlowTemplateModelWithPropertySetters:
```
Rename to 
```suggestion
flow CreateFlowTemplateModelWithPropertySetters_UsingParsedFyPyFileAndPropertySettersTemplateModels -> FlowTemplateModelWithPropertySetters:
```

**Review by PerisicJere — APPROVED** [review]

**Pull request merged**